### PR TITLE
134 - Strict types for arguments and return types

### PR DESCRIPTION
**Pull request by lutdev**
I made the code more strict

Issue - https://github.com/matomo-org/matomo-php-tracker/issues/134

### CODE REVIEW

File: MatomoTracker.php [hunk 1]
```diff
@@ -248,12 +248,12 @@ public function setApiUrl(string $url)
      * It is recommended to only send UTF-8 data to Matomo.
      * If required though, you can also specify another charset using this function.
      *
-     * @param string $charset
      * @return $this
      */
-    public function setPageCharset($charset = '')
+    public function setPageCharset(string $charset = ''): self
```
Comment (sgiehl) [inline MatomoTracker.php:253]: Typehinting `self` will make it impossible to overwrite a method in an extending class in PHP < 7.4
So would be better to not typehint that until we change the php requirements.

Comment (lutdev) [reply MatomoTracker.php]: @sgiehl agree, good point. Thank you, deleted `: self` everywhere.

**Comment by lutdev** [comment]
@sgiehl just a friendly reminder about PR :)

**Review by sgiehl — CHANGES_REQUESTED** [review]
Hey @lutdev. Sorry for the delay. Had a lot other stuff to handle. The changes overall already look fine to me. I nevertheless added a comment we could consider to change. It might though not be needed for this PR. If you want we can also merge this and change the return values in a new PR.

File: MatomoTracker.php [hunk 2]
```diff
@@ -237,7 +237,7 @@ public function __construct($idSite, $apiUrl = '')
         $this->visitorCustomVar = $this->getCustomVariablesFromCookie();
     }
 
-    public function setApiUrl(string $url)
+    public function setApiUrl(string $url): void
```
[the remark this answers is not in the document]
Comment (lutdev) [reply MatomoTracker.php:240]: @sgiehl I agree with you and it was in mind but for the next PR) So, if it's okay for you after merge I will create new issue and describe upcoming changes.

**Pull request merged**